### ISSUE TITLE
Callback when dragged file leaves the drag&drop area

### ISSUE THREAD

**Issue by orangesoup** [danialfarid/ng-file-upload#1043]
Is it possible to call a function when the dragged file actually leaves the area? I would like to change a variable when the file is over the area and then change it back when the drop is canceled for some reason. I know I can call a function with `ngf-drag-over-class` and set the variable, but I don't know how to change it back on cancel.

Am I just missing it, or there is no way to do this?

**Comment by danialfarid** [comment]
You can register your own event listener or if the variable that you set is just used for displaying and css purposes you can use the dragover class.

to register you listeners
```
app.directive('dragLeave, function() {
  return {
    scope: true,
    link: function(scope, element, attr) {
      element[0].addEventListener('dragleave', function() {

      }, false);
    }
  };
});
```

I will add this feature to the next version.

**Comment by orangesoup** [comment]
Thanks!

I have an image in the drop area which is grey by default. I'm using angular material so I thought I'll take advantage of `md-accent` etc. My goal is to set variable `x`  to `true/false` and then with `ng-class` set `md-accent` via that variable. So on dragover the image would be pink, otherwise grey. 

I know it's possible to just set the color with css like `dragover svg { fill: pink; }`, but then the whole theming of Angular Material would be pointless.

Also, not really connecting, just don't know if I should open an another issue for this: if there's any element in the drag area I'll have to use `pointer-events: none` on all of them, otherwise while dragging the object these elements would flicker. Is this the intended behavior?

**Comment by danialfarid** [comment]
For the error if you create a jsfiddle I can look into it. Try to set the `dragOverDelay: 100` that might solve the issue

**Comment by danialfarid** [comment]
Fixed at 9.0.5.
The delay is by default 100 now so it should fix the flickering issue without setting delay.
Use `ngf-drag($isDragging)` for the first issue.

**Comment by orangesoup** [comment]
Thanks for the fix! Delay 100 wouldn't solve the problem, but actually I managed to narrow it down a bit. On Windows with Chrome latest (v45) there is no error, it happens on Ubuntu with Chrome though (latest version as well.)

I made a  [jsfiddle](http://jsfiddle.net/99hcc7v7/1/) anyway, but if it's only happening on Ubuntu I don't really care. :)
